app/views/dishes/show.html.haml

```diff
@@ -23,5 +23,6 @@
       %p
       - if can? :create, ShoppingCart
         = link_to 'Add to cart', dish_add_item_path(@dish), method: :post, class: 'btn btn-success'
-      - if can? :manage, Dish
+      - if (can? :manage, Dish) && (current_user == @dish.restaurant.user)
```

Comment:
`can? :manage, Dish` should be enough, depending on how that ability is configured. Have a look at that before adding conditions